### PR TITLE
#131 `locks.lock()` must check whether the job is finished

### LINKED ISSUE #131: `locks.lock()` must check whether the job is finished
The `locks.lock()` method must check whether there exist any jobs under the given name, which are not yet finished (and not expired). If such jobs exist, the lock should not be granted.

### PR DESCRIPTION
**Pull request by Suban05**
## Changes

* Added the checking of whether the job is finished, using the method `jobs.busy?`
* Deleted lock/unlock in the pipeline, because we introduced a new check into the locking mechanism
* Fixed tests: finish jobs before calling methods from `front_push`, which calls `locks.lock`

Fixes #131

### CODE REVIEW

**Comment by Suban05** [comment]
@Yegorov can you review it, please?

File: test/baza/test_locks.rb [hunk 1]
```diff
@@ -45,4 +46,19 @@ def test_simple_locking_scenario
...
+    name = "#{fake_name}-a"
+    job = token.start(name, fake_name, 1, 0, 'n/a', [], '192.168.1.1')
+    assert(human.jobs.busy?(name))
+    owner = "baza #{Baza::VERSION} #{Time.now.utc.iso8601}"
+    assert_raises(Baza::Locks::Busy) do
+      human.locks.lock(name, owner)
+    end
+    job.finish!(fake_name, 'stdout', 0, 544, 1, 0)
+    human.locks.lock(name, owner)
+    assert human.locks.locked?(name)
```
Comment (Yegorov) [inline test/baza/test_locks.rb:62]: @Suban05 Most likely, parentheses were missed here. Better to have one style
```ruby
assert(human.locks.locked?(name))
```

**Review by Yegorov — CHANGES_REQUESTED** [review]

**Comment by Suban05** [comment]
@Yegorov fixed

**Review by Yegorov — APPROVED** [review]
All good!

**Comment by Suban05** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Suban05 something is wrong with the branch?

**Comment by Suban05** [comment]
@yegor256 There were no such errors on the local machine. I'll see what might be wrong

**Comment by yegor256** [comment]
@Suban05 maybe timeouts are too small, this may be the problem

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
@Yegorov Thanks for your code review! Here's how your +15 points were calculated: +25 base points, +0 for self-review, +0 for hits-of-code (< 15 HOC), -10 for few comments (4), and +0 for comment bonus (max 20). Your [running balance is now +146](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). Keep up the good work and aim for more detailed reviews next time! 💻👍

**Comment by 0crat** [comment]
@Suban05 Thanks for the contribution! You've earned +25 points. Please, [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) them coming. Your running balance is [+262](https://zerocracy.github.io/judges-action/zerocracy-vitals.html).

**Pull request merged**